hooks/firebird/use-event-stream.tsx

```diff
@@ -105,14 +110,47 @@ const useDeploymentEventStream = (deploymentId: string) => {
       if (event instanceof MessageEvent) {
         const data: DeploymentStreamMessage = JSON.parse(event.data)
         console.log("SSE: Client received a message", data)
-
         updateDeploymentProgress(data)
         updateDeploymentStatus(data)
       }
     },
     [updateDeploymentProgress, updateDeploymentStatus],
   )
 
+  // To keep method consistent with the event stream and avoid unnecessary call errors
+  useEffect(() => {
+    if (deploymentStatus === DeploymentStatus.STARTING) {
+      setLastDeploymentStatus(DeploymentStatus.STARTING)
+    } else if (
+      deploymentStatus === DeploymentStatus.STOPPING ||
+      deploymentStatus === DeploymentStatus.STOPPED
+    ) {
+      setLastDeploymentStatus(DeploymentStatus.STOPPING)
+    } else if (deploymentStatus === DeploymentStatus.DEPLOYING) {
+      setLastDeploymentStatus(DeploymentStatus.DEPLOYING)
+    }
+  }, [deploymentStatus])
+
+  const getMethod = (): string => {
+    if (deploymentStatus === DeploymentStatus.STARTING) {
+      return "follow_deployment_start"
+    } else if (
+      deploymentStatus === DeploymentStatus.STOPPING ||
+      deploymentStatus === DeploymentStatus.STOPPED
+    ) {
+      return "follow_deployment_stop"
+    } else if (deploymentStatus === DeploymentStatus.DEPLOYING) {
+      return "follow_deployment"
+    } else if (deploymentStatus === DeploymentStatus.ACTIVE) {
+      return lastDeploymentStatus === DeploymentStatus.STARTING
+        ? "follow_deployment_start"
+        : "follow_deployment"
+    }
+    return "follow_deployment"
```

Comment:
Could you put this into something more readable like a switch statement? (Also, switch wouldn't require you to go through each case)

https://builtin.com/articles/switch-case-vs-if-else